autoload/wiki/complete.vim

```diff
@@ -207,7 +212,8 @@ endfunction
 
 function! s:completer_tags.complete(regex) dict abort " {{{2
   let l:candidates = keys(wiki#tags#get_all())
-  call filter(l:candidates, {_, x -> stridx(x, a:regex) >= 0})
+  call filter(l:candidates, {_, x -> match(x,
+        \ (g:wiki_completion_case_sensitive ? '\C' : '\c') . a:regex) >= 0})
```

Comment:
I notice the pattern is repeated several times. Perhaps it would look better if we abstracted into a separate function somehow? E.g. `call s:filter(l:candidates, a:regex)` where `s:filter` does the necessary work? This could also help slightly improve things by "caching" the `g:wiki_completion_case_sensitive` choice (sorry, premature optimization again; I can't help it).

Reply:
Caching the case sensitive choice would break some of your tests, though, so I would probably not care about that. But simplifying the pattern by abstracting it to a separate filter function seems possibly useful to me.

Reply:
I think the only place that would break down is on line 32 where the comparison is `== 0` rather than `>= 0`. I’ll give a try breaking it out into a function for the other lines and add a commit if it seems good.

I’d argue in general against caching `g:` options; from a usability standpoint, it’s better to allow an option to change whenever the user wants to, rather than requiring a restart.